File: chrome/browser/android/messages_resource_mapper_initializer.cc
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/android/messages/jni_headers/MessagesResourceMapperInitializer_jni.h"
#include "chrome/browser/android/resource_mapper.h"
#include "components/messages/android/message_dispatcher_bridge.h"

// Sets up a callback for MessageDispatcherBridge additional initialization. For
// example, the ResourceMapper from chrome can be bound for use in components.
void JNI_MessagesResourceMapperInitializer_Init(JNIEnv* env) {
  messages::MessageDispatcherBridge::Get()->Initialize(
      base::BindRepeating(&ResourceMapper::MapToJavaDrawableId));
}

File: chrome/browser/android/messages/java/src/org/chromium/chrome/browser/messages/MessagesResourceMapperInitializer.java
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.messages;

import org.jni_zero.NativeMethods;

/** Helper class to initialize ResourceIdMapper for MessageDispatcherBridge. */
public class MessagesResourceMapperInitializer {
    /** Calls native method to initialize ResourceIdMapper for MessageDispatcherBridge. */
    public static void init() {
        MessagesResourceMapperInitializerJni.get().init();
    }

    @NativeMethods
    interface Natives {
        void init();
    }
}
